### PR TITLE
Streaming reporters and dumpers

### LINKED ISSUE #132: Report processing runs out of memory with large report file.
Report file was 1.5GB, physical memory on this instance is about 7GB. Anything I can do to successfully process this report?

### PR DESCRIPTION
**Pull request by tsenart**
This changeset makes vegeta reporters and dumpers work with bounded
memory by replacing batch algorithms with streaming counterparts.

Fixes #132

### CODE REVIEW

**Comment by tsenart** [comment]
PTAL: @ohookins, @xla, @cwinters, @peterbourgon, @Mistobaan

File: lib/dumpers.go [hunk 1]
```diff
@@ -1,43 +1,34 @@
...
-type Dumper interface {
-	Dump(*Result) ([]byte, error)
-}
-
-// DumperFunc is an adapter to allow the use of ordinary functions as
-// Dumpers. If f is a function with the appropriate signature, DumperFunc(f)
-// is a Dumper object that calls f.
-type DumperFunc func(*Result) ([]byte, error)
+// CSVDumper implements the Reporter interface by reporting a Result as a CSV
+// record with six columns. The columns are: unix timestamp in ns since epoch,
```
Comment (peterbourgon) [inline lib/dumpers.go]: UNIX :)

File: lib/dumpers.go [hunk 2]
```diff
@@ -1,43 +1,34 @@
...
+// http status code, request latency in ns, bytes out, bytes in, and lastly the error.
```
Comment (peterbourgon) [inline lib/dumpers.go]: HTTP :)

File: lib/dumpers.go [hunk 3]
```diff
@@ -1,43 +1,34 @@
...
-	_, err := fmt.Fprintf(&buf, "%d,%d,%d,%d,%d,\"%s\"\n",
-		r.Timestamp.UnixNano(),
-		r.Code,
-		r.Latency.Nanoseconds(),
-		r.BytesOut,
-		r.BytesIn,
-		r.Error,
+// Report partially implements the Reporter interface.
+func (d CSVDumper) Report(w io.Writer) error {
+	_, err := fmt.Fprintf(w, "%d,%d,%d,%d,%d,\"%s\"\n",
```
Comment (peterbourgon) [inline lib/dumpers.go]: `\"%s\"` is better spelled `%q`

**Comment by xla** [comment]
LGTM

**Comment by tsenart** [comment]
Blocked by https://github.com/streadway/quantile/pull/4

**Pull request merged**